backend/src/main/java/fk/prof/backend/model/request/RecordedProfileIndexes.java

```diff
@@ -0,0 +1,45 @@
+package fk.prof.backend.model.request;
+
+import com.koloboke.collect.map.hash.HashIntObjMap;
+import com.koloboke.collect.map.hash.HashIntObjMaps;
+import com.koloboke.collect.map.hash.HashLongObjMap;
+import com.koloboke.collect.map.hash.HashLongObjMaps;
+import recording.Recorder;
+
+import java.util.List;
+
+public class RecordedProfileIndexes {
+  private final HashLongObjMap<String> methodLookup = HashLongObjMaps.newUpdatableMap();
+  private final HashIntObjMap<String> traceLookup = HashIntObjMaps.newUpdatableMap();
+
+  public String getMethod(long methodId) {
+    return methodLookup.get(methodId);
+  }
+
+  public String getTrace(int traceId) {
+    return traceLookup.get(traceId);
+  }
+
+  public void update(Recorder.IndexedData indexedData) {
+    updateMethodIndex(indexedData.getMethodInfoList());
+    updateTraceIndex(indexedData.getTraceCtxList());
+  }
+
+  private void updateMethodIndex(List<Recorder.MethodInfo> methods) {
+    if (methods != null) {
+      for (Recorder.MethodInfo methodInfo : methods) {
+        methodLookup.put(methodInfo.getMethodId(), String.format("%s.%s%s",
+            methodInfo.getClassFqdn(), methodInfo.getMethodName(), methodInfo.getSignature()));
```

Comment:
Ah and that too by a large margin. Will replace this with a stringbuilder here, although string.format in logger methods (or others which are infrequently called) can be kept around, since the code using it is more readable.